### ISSUE TITLE
Problems with Circles

### ISSUE THREAD

**Issue by McNugget6750** [fablabnbg/inkscape-chain-paths#3]
Hi,

First off: Please be aware that your little piece of software is THE HOLY GRALE for many many people that work with dxf files and it needs more traction and get more publicity or it needs to become part of inkscape itself!

On with the problem:
I export a dxf file out of solidworks and I'm using your tool to merge all the separated paths to make one path that actually works for milling or in my case laser cutting.

I encountered a problem with circles. Take this group of circles for example:
![capture0](https://cloud.githubusercontent.com/assets/12099353/20511211/05d55ac0-b02b-11e6-9308-236336654dc5.PNG)

As you can see when I pull on the lines of the circle, the circle is fully connected (warning, sometimes the nodes are not set to "smooth" right after dxf import, so they come in in multiple variants) and the circle consists of one line all around.
![capture-1](https://cloud.githubusercontent.com/assets/12099353/20511223/327deea2-b02b-11e6-93c9-ba80831aaa55.PNG)

Now I apply the chain extension to the whole thing and I see this! There are some extra lines that make no sense and deleting them individually does not solved the irregular path the laser tries to cut.
![capture1](https://cloud.githubusercontent.com/assets/12099353/20511213/0790cc28-b02b-11e6-8ac8-ac0a6324fda8.PNG)

It becomes more obvious when I pull the paths apart because then you see this:
![capture2](https://cloud.githubusercontent.com/assets/12099353/20511214/08c36ace-b02b-11e6-8e04-ec65f64aba0d.PNG)

I believe this is a bug but I'm not sure where it is. I use InkScape 0.91 and I really hope we can find and eliminate this bug because your plugin is the missing extension that people need to successfully work with DXF files and open source laser cutters! Here is the dxf file for your reference, in case you want to try it yourself with my file.
[Hood-2ndLayer.zip](https://github.com/fablabnbg/inkscape-chain-paths/files/605518/Hood-2ndLayer.zip)

Cheers,
Timo

**Comment by jnweiger** [comment]
I agree, it should become part of inkscape, or at least one of the standard places where people look for inkscape extensions.
As to the issue: Yes, I've seen similar effects with inkscape (unrelated to that extension) -- hard to reproduce so far. Maybe your DXF helps someone to pinpoint the issue. Thanks!
